modules/components/src/DataTable/ColumnsState.js

```diff
@@ -126,6 +126,18 @@ export default class extends Component {
     this.setState({ temp }, () => this.save(temp));
   };
 
+  add = column => {
+    const { id } = column;
+    let existing = this.state.config.columns.find(x => x.id === id);
+    if (existing) return;
+    let temp = {
+      ...this.state.config,
+      columns: this.state.config.columns.concat({ ...column }),
```

Comment:
tempted to say `[... this.state.config.columns, { ...column }]` but not a big thing :P